email-service/internal/mail/backends/gomail.go

```diff
@@ -46,11 +57,18 @@ func (gm *GomailMailer) SendEmail(
 
 	select {
 	case <-ctx.Done():
-		return fmt.Errorf("email sending cancelled: %w", ctx.Err())
+		err := ctx.Err()
+		gomailErrorsCounter.Inc()
+		getLogger().Error("context done", slog.Any("error", err))
+		return fmt.Errorf("email sending cancelled: %w", err)
 	case err := <-done:
 		if err != nil {
+			gomailErrorsCounter.Inc()
```

Comment:
we should ask Pavlo: is it okay to call metrics directly from the business code? Maybe it's better to implement a metrics facade? This way it will abstract you from the specific implementation, in case you decide to switch to prometheus from victoria metrics